iceoryx_utils/test/moduletests/test_cxx_function_ref.cpp

```diff
@@ -0,0 +1,228 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iceoryx_utils/cxx/function_ref.hpp"
+#include "test.hpp"
+
+using namespace ::testing;
+using namespace iox::cxx;
+
+int freeFunction()
+{
+    return 42 + 42;
+}
+
+class Functor
+{
+  public:
+    int operator()()
+    {
+        return 73;
+    }
+};
+
+struct ComplexType
+{
+    char a;
+    int b;
+    float c;
+
+    bool operator==(const ComplexType& rhs) const
+    {
+        if (a == rhs.a && b == rhs.b && c == rhs.c)
```

Comment:
Why not write simply:
```
return ( a == rhs.a && b == rhs.b && c == rhs.c);
```
This would save you 7 lines and lines these days are very expensive ;)

Reply:
that's boring ... on the other side, Walter E. Brown knows much more interesting ways to return true ;)

Reply:
If it is test code I don't care about LoC, but prefer to make it more readable :-) For everyone interested in the brilliant talk, here's the link: https://www.youtube.com/watch?v=OAmWHmwlMwI

Reply:
I disagree. How does it make your code more readable when you just add more lines?
The first line ```( a == rhs.a && b == rhs.b && c == rhs.c)``` have to be understood either way and with your approach you have 6 more lines to understood